packages/node-core/src/indexer/postgresBasedDb.ts

```diff
@@ -0,0 +1,142 @@
+// Copyright 2020-2022 OnFinality Limited authors & contributors
+// SPDX-License-Identifier: Apache-2.0
+
+import {Sequelize, DataTypes, Model, ModelStatic} from 'sequelize';
+import {getLogger} from '../logger';
+
+const logger = getLogger('pg-based-db');
+
+interface NodeMap {
+  [key: string]: Buffer;
+}
+
+interface KVStoreAttributes {
+  key: number;
+  value: Buffer;
+}
+
+class KVStoreInstance extends Model<KVStoreAttributes> implements KVStoreAttributes {
+  key!: number;
+  value!: Buffer;
+}
+
+interface LeafLengthAttributes {
+  id: number;
+  value: number;
+}
+
+class LeafLengthInstance extends Model<LeafLengthAttributes> implements LeafLengthAttributes {
+  id!: number;
+  value!: number;
+}
+
+export class PgBasedDB {
```

Comment:
We can implement this interface https://github.com/subquery/subql/blob/main/packages/node-core/src/merkle-mountain-range.d.ts#L18

Reply:
The names of everything need to be much clearer. `PgBasedDB` has no indication this relates to MMR, same with `KVStore`